//
// Juniper P4 Agent
//
/// @file  jp4cli.proto
/// @brief CLI service for JP4Agent
//
// Created by Karthikeyan Sivaraj, February 2018
// Copyright (c) [2018] Juniper Networks, Inc. All rights reserved.
//
// All rights reserved.
//
// Notice and Disclaimer: This code is licensed to you under the Apache
// License 2.0 (the "License"). You may not use this code except in compliance
// with the License. This code is not an official Juniper product. You can
// obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
//
// Third-Party Code: This code may depend on other components under separate
// copyright notice and license terms. Your use of the source code for those
// components is subject to the terms and conditions of the respective license
// as noted in the Third-Party source code file.
//

syntax = "proto3";

package jp4cli;

service CmdHandler {
    rpc SendCmd (CmdRequest) returns (CmdReply) {}
}

message CmdRequest {
    string cmdstr = 1;
}

message CmdReply {
    int32  status = 1;
    string cmdout = 2;
}
